### PR TITLE
Determine feed positions of PCO using PCOAntenna 

### PR DESCRIPTION
**Pull request by lazdam**
See linked issue: #31

### CODE REVIEW

File: ch_util/tools.py [hunk 1]
```diff
@@ -783,8 +827,24 @@ def find(name):
 
         # Dealing with a HCRO feed
 
-        # Temporary setting until this is defined
-        pos = None
+        # Determine position
```
Comment (jrs65) [inline ch_util/tools.py:830]: Please refactor this to not just copy the exact same code block 4 times.

**Review by jrs65 — CHANGES_REQUESTED** [review]
Hi, this is mostly looking good. A few requests:

- As noted in the comments please can you refactor this to not just have the same block repeated 4 times in the routine? Either do it once and reuse it, or pull it out into a function.
- Please follow our code guidelines (https://github.com/chime-experiment/Pipeline/blob/master/CONTRIBUTING.md#rules). Generally we like to see one well formatted commit per logical (and externally visible) change. Commits should apply linearly, i.e. no merge commits, so please rebase and force push to this branch if needs be.

File: ch_util/tools.py (hunk 1 above)
Comment (aaronpearlman) [reply ch_util/tools.py:830]: @lazdam Would you be willing to put the feed position calculation for Outriggers into a separate function outside of ` _get_input_props`, and then call it for each Outrigger antenna inside ` _get_input_props` (as per @jrs65's request)?

Also, we need to lint the code. It did not pass the test.

Comment (jrs65) [reply ch_util/tools.py:830]: I think the block for the CHIME antenna is the same too, so please include that as well.

Comment (lazdam) [reply ch_util/tools.py:830]: It's almost the same for CHIME, though slightly different. E.g. for CHIME, we have: 
```      
if "y_offset" not in lay.node_property(slt):
   pos[1] += (float(slt.sn[-1]) - 1.5) * 0.3048
```
whereas for the outriggers we have
```      
if "y_offset" not in lay.node_property(slt):
   pos[1] += (float(slt.sn[-1]) - 0.5) * 0.3048
```
(only difference being -1.5 vs. -0.5). In the function that I write, I can add a parameter that takes care of this, e.g.: 

```
def _get_feed_positions(lay, rfl, foc, cas, slt, slot_factor):
        try:
            pos = [0.0] * 3

            for node in [rfl, foc, cas, slt]:
                prop = lay.node_property(node)

                for ind, dim in enumerate(["x_offset", "y_offset", "z_offset"]):

                    if dim in prop:
                        pos[ind] += float(prop[dim].value)  # in metres

            if "y_offset" not in lay.node_property(slt):
                pos[1] += (float(slt.sn[-1]) - slot_factor) * 0.3048

        except:
            pos = None
        
        return pos
```
where we have the `slot_factor` that gets set to 1.5 when `cyl` \in [2,5], and 0.5 for the outrigger cylinders. If there's a cleaner way to do this, let me know.

Comment (kiyo-masui) [reply ch_util/tools.py:830]: Yeah, that is exactly what I would do. No need to overcomplicate it, as simple function like the above is perfect.

Comment (aaronpearlman) [reply ch_util/tools.py:830]: Sure, that works. Please put some documentation (e.g., a brief comment in the code) that `slot_factor` is `1.5` for CHIME and `0.5` for Outriggers, so we don't forget the meaning of this parameter.

File: ch_util/tools.py [hunk 2]
```diff
@@ -154,6 +154,20 @@
 _PF_ROT = 1.986  # Pathfinder rotation from north (towards west) in degrees
 _PF_SPACE = 22.0  # Pathfinder cylinder spacing
 
+# PCO geometry
+_PCO_POS = [0.0, 0.0, 0.0]
+_PCO_ROT = (
+    -0.67
```
Comment (lazdam) [inline ch_util/tools.py:160]: Not sure why black decided to format it this way, will try to fix it.

File: ch_util/tools.py (hunk 1 above)
Comment (lazdam) [reply ch_util/tools.py:833]: Ok, done. I also squashed everything into 1 commit and forced pushed. I didn't know that was a thing, though I enjoy how clean the commit is and having all the changes on one file.

**Comment by lazdam** [comment]
Is there any way to check which version of `black` we're using? I have the latest, so if that causes the lint-code test to fail, I'll have to swap versions.

**Comment by aaronpearlman** [comment]
@lazdam I just approved and ran the checks. Everything passed. The lint checker is running with `black-22.8.0`. This looks good to me!

I think we're ready to merge it in once @jrs65 approves it.

**Review by jrs65 — APPROVED** [review]

**Review by jrs65 — CHANGES_REQUESTED** [review]
Sorry @lazdam, one quick last thing, can you change the commit message to follow the guidelines. Probably should be `feat(tools): enable feed position calculation for outriggers`

**Comment by lazdam** [comment]
> Sorry @lazdam, one quick last thing, can you change the commit message to follow the guidelines. Probably should be `feat(tools): enable feed position calculation for outriggers`. It's important for our changelog generation to have this format.
> 
> Also, please rebase on the latest master while you do so.
> 
> Otherwise, it looks great. Thanks! When you've made the changes I can merge immediately.

Should be done. Rebased with master and updated commit message.

**Comment by aelanman** [comment]
@jrs65 Is there anything more that needs to be done for this to be merged?

**Review by jrs65 — APPROVED** [review]

**Comment by aelanman** [comment]
Thanks @jrs65 !

**Pull request merged**